bindings/generate_bindings.py

```diff
@@ -1,341 +1,313 @@
+    return f"    {param.name}: {typestr}{default},"
 
-    def load(self, val: Code) -> Code:
-        return f"{self.name}.from_json({val})"
 
-    def dump(self, val: Code, omit_unset: Code) -> Code:
-        return f"{val}.to_json({omit_unset})"
-
-    def gen_def(self) -> Code:
-        required = sorted(p for p in self.params if self.params[p].required)
-        optional = sorted(p for p in self.params if not self.params[p].required)
-        out = [f"class {self.name}:"]
-        for k in optional:
-            v = self.params[k]
-            out += [f'    {k}: "typing.Optional[{v.type.annotation(prequoted=True)}]" = None']
+def is_streaming_response(defn: typing.Optional[swagger_types.TypeDef]):
+    if isinstance(defn, swagger_types.Class) and set(defn.params.keys()) == set(
+        ["result", "error"]
+    ):
+        error_ref = defn.params["error"]
+        return (
+            isinstance(error_ref.type, swagger_types.Ref)
+            and error_ref.type.name == "runtimeStreamError"
+        )
+    return False
+
+
+def unwrap_streaming_response(anno: swagger_types.TypeAnno):
+    if not isinstance(anno, swagger_types.Ref) or not is_streaming_response(anno.defn):
+        return anno
+    assert isinstance(anno.defn, swagger_types.Class)
+    return anno.defn.params["result"].type
+
+
+def gen_function(anno: swagger_types.Function) -> Code:
+    # Function name.
+    out = [f"def {anno.method}_{anno.name}("]
+
+    # Function parameters.
+    out += ['    session: "api.Session",']
+    if anno.params:
+        out += ["    *,"]
+
+    required = sorted(
+        ((pname, param) for pname, param in anno.params.items() if param.required),
+        key=head,
+    )
+    optional = sorted(
+        ((pname, param) for pname, param in anno.params.items() if not param.required),
+        key=head,
+    )
+    for _, param in required + optional:
+        out += [gen_function_param(param)]
+
+    # Function return type.
+    # We wrap the return type annotation for streaming or union responses.
+    need_quotes = anno.streaming or len(anno.responses) > 1
+    if anno.streaming:
+        anno.responses = {code: unwrap_streaming_response(r) for code, r in anno.responses.items()}
+    returntypes = set(annotation(r, prequoted=need_quotes) for r in anno.responses.values())
+    returntypestr = ",".join(sorted(returntypes))
+    if len(returntypes) > 1:
+        returntypestr = f"typing.Union[{returntypestr}]"
+    if anno.streaming:
+        returntypestr = f"typing.Iterable[{returntypestr}]"
+    if need_quotes:
+        returntypestr = f'"{returntypestr}"'
+
+    out += [f") -> {returntypestr}:"]
+
+    # Function body.
+    has_path_params = any(p for p in anno.params.values() if p.where == "path")
+    # body_params = sorted(p for p in self.params if anno.params[p].where == "body") # not in use
+    query_params = sorted(
+        ((pname, param) for pname, param in anno.params.items() if param.where == "query"),
+        key=head,
+    )
```

Comment:
This is equivalent to:

```python
query_params = sorted((pname, param) for pname, param anno.params.items()) if param.where == "query"))
```
That is, `sorted` automatically sorts tuples by their first element.  I think the default tuple sort then falls back to the second element, which is different than your `head` function, but also not materially different in this case, since your `pnames` are guaranteed to be unique here.

I think every instance of `key=head` in this file can be omitted.  And you can fit the `sorted()` calls on a line if you use `(k, v) for k, v in anno.params.items()`, since `k` and `v` are common enough pneumonics for "key" and "value".